File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.*;
import java.util.stream.Collectors;
class Solution {
    public List solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        List <Integer> list=new ArrayList<>();
        List <Integer> progress= Arrays.stream(progresses).boxed().collect(Collectors.toList());
        List<Integer> speed=Arrays.stream(speeds).boxed().collect(Collectors.toList());
        int days=(100-progress.get(0))%speed.get(0)==0?(100-progress.get(0))/speed.get(0):(100-progress.get(0))/speed.get(0)+1;
        

        int tmp=0;
        while(!progress.isEmpty()){
            int finishDay=(100-progress.get(0))%speed.get(0)==0?(100-progress.get(0))/speed.get(0):(100-progress.get(0))/speed.get(0)+1;
            if(finishDay<=days){
                tmp++;
            }
            else{
                list.add(tmp);
                days=finishDay;
                tmp=1;
            }
            if(progress.size()==1){
                list.add(tmp);
            }
            System.out.println(progress.get(0));
            progress.remove(progress.get(0));
            speed.remove(speed.get(0));
        }
        return list;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    arr = list(zip(progresses, speeds))
    finish_day = []
    for p in arr:
        finish_day.append(checkFinishDay(p[0], p[1]))
    #print(finish_day)
    
    day = finish_day[0]
    finish = 1
    for i in range(1, len(finish_day)):
        if day >= finish_day[i]:
            finish += 1
        else:
            answer.append(finish)
            day = finish_day[i]
            finish = 1
    answer.append(finish)
    return answer

def checkFinishDay(progress, speed):
    day = 0
    while progress < 100:
        progress += speed
        day += 1
    return day